.github/workflows/ci-sglang-benchmark.yml

```diff
@@ -69,21 +69,135 @@ jobs:
           # Try with the latest nightly releases, not what iree-turbine pins.
           # We could also pin to a known working or stable version.
           # This should eventually stabilize. Do the best we can for now.
-          pip install -f https://iree.dev/pip-release-links.html --upgrade \
-            iree-base-compiler==3.0.0rc20241118 \
-            iree-base-runtime==3.0.0rc20241118 \
+          pip install -f https://iree.dev/pip-release-links.html --upgrade --pre \
+            iree-base-compiler \
+            iree-base-runtime \
             "numpy<2.0"
 
       - name: Install SGLang
         run: pip install "git+https://github.com/nod-ai/sglang.git#subdirectory=python"
 
-      - name: Launch Shortfin Server
-        run: pytest -v app_tests/benchmark_tests/llm/sglang_benchmarks/sglang_benchmark_test.py --log-cli-level=INFO --html=out/llm/sglang/index.html
+      - name: Run Shortfin Benchmark Tests
+        run: pytest -v app_tests/benchmark_tests/llm/sglang_benchmarks/shortfin_benchmark_test.py --log-cli-level=INFO --html=out/llm/shortfin/index.html
+
+      - name: Deploy to GitHub Pages
+        uses: peaceiris/actions-gh-pages@4f9cc6602d3f66b9c108549d475ec49e8ef4d45e # v4.0.0
+        with:
+          github_token: ${{ secrets.SHARK_PLATFORM_GH_TOKEN }}
+          publish_dir: ./out/llm/shortfin
+          destination_dir: ./llm/sgl_benchmark/shortfin
+          keep_files: true
+
+  benchmark_sglang:
+    if: ${{ github.repository_owner == 'nod-ai' || github.event_name != 'schedule' }}
+    name: "SGLang Serving Benchmark With SGLang"
```

Comment:
We could theoretically move the SGLang benchmarking part to https://github.com/nod-ai/sglang.

The downside of that would be that it would fragment the benchmark results across repos. Would need to check actions across different repos to be able to update the benchmark numbers daily. Would also make it so that the SGLang benchmark results wouldn't be included in our pytest dashboard. We have plans to upload the stats to Grafana, which we can start working on once this half of the benchmark is in. Adds some complexity maintaining across different repos, but shouldn't be a crazy amount of complexity. I'm not sure what we would do if we move our `Shortfin` backend from our forked repo to the origin SGLang repo. I assume in that case, we wouldn't be able to carry our own benchmarking & upload process.

We could also make another repo for it. I would need some help getting that repo created and runner access set up for it, since I don't have that level of permission. That would have some of the fragmentation issues as well, not being included in the shark-ai pytest dashboard. Would be easier to keep the `grafana` stuff organized though. Maybe we could just add a link to that dashboard in our top-level `index.html` file in this repo, which would mitigate the fragmentation issue for both approaches in regards to pytest dashboard.

Ultimately, I was thinking that having the code here follows the "pattern" of other benchmark tests more closely and would keep things centralized. SGLang code doesn't rely on `shortfin`, but the benchmark itself is a target for shortfin. I `think` that having the benchmarks as a nightly run and using mi300x-4 for only the sglang integration and benchmark tests would mitigate strain on infrastructure.

Which option do you like the best, given above statements?

Reply:
We don't need to tightly couple

* the report data generation (project code, test code, workflow files)
* the report data itself (structured json output or standalone .html report files)
* the dashboard presentation layer (currently gh-pages)

I consider the current dashboard existing at https://nod-ai.github.io/shark-ai/ to be a v0 prototype and I don't want to bend the repository structure by treating it as a final implementation. We could be reaching the point here where we need a more complete design (including a review process).

Some random notes from looking there:

* We could also use something like https://pypi.org/project/pytest-html-report-merger/ or https://pypi.org/project/pytest-html-merger/ to group test reports on that dashboard
* It looks like the current process overwrites previous results, e.g. https://github.com/nod-ai/shark-ai/commit/f7d53452dbd57debdcbdaa125da221dd3a4ed6f4 . Historical result tracking would be useful too: a dropdown to view over results, metrics like number of tests passing / xfailed over time, etc.

Maybe we can keep this complexity local to https://github.com/nod-ai/shark-ai/tree/main/app_tests/ and a few appropriately named github workflows (as you have it here, but with a clearer definition of the "core project" and "external integration tests"). That seems to be a reasonable starting point / compromise to me, as long as we draw a hard line there and stick to it.

Reply:
Yeah, this makes a ton of sense and clears up the flow for me. What I'm thinking is:

## Reaching the boundaries of using the pytest dashboard
I think `Grafana` will be a good solution for this. We can get it to a point where the dashboard (if it sticks around) is more just to quickly see the status of a test, where Grafana would be used for actually tracking metrics. When I work on that (as next task), I can write it in a general way to make it easy for other benchmark tests to transition.

## Merging pytest reports
This is useful, I'll test locally and see which one fits the bill

## Isolating complexity
Yeah, I agree, what you described is a good starting point, but not the end-all-be-all. 

It makes sense to have a dedicated benchmarking repo as a more over-arching project, especially since we've been expanding benchmarks a good bit lately, and #609 transitions them to using packages, so they don't all NEED to run in `shark-ai` itself. I'll add a description at the top of this CI file adding more detail and explaining that this is an `external test`, which should not be copied in most cases or any case outside of `app_tests`. We should continue discussions on a dedicated repo though. It hasn't been formally discussed, but it would be a logical continuation to add benchmarking against more external projects in the future, like [TensorRT](https://developer.nvidia.com/tensorrt), which was mentioned in lead-sync today as faster than SGLang

Reply:
Added a `README` description to top of CI file, explaining the details of why we're including it here, for now. Leaving conversation unresolved, in case you have further suggestions

Reply:
Added steps to merge reports and upload as one artifact to pytest dashboard